test/models/parser/page_item_test.rb

```diff
@@ -539,4 +516,15 @@ def fake_twitter_user
     oembed_url = Parser::PageItem.new('https://example.com').oembed_url(oembed_doc)
     assert_equal "https://example.com/oembed", oembed_url
   end
+
+  test "shouldn't break when cannot get twitter url" do
+  # I'm trying to write a substitute to "should handle error when cannot get twitter url"
```

Comment:
for this test, I think we should attempt to model what happens if the request actually errors, or returns a response code we dislike. I could see it actually being a couple of tests:
1. mocking our request to Twitter to return an error code and message that is a 422, 500, etc. basically anything that isn't a 200 or 301 (<400, which is how we have it written in the implementation). then could test that the error gets sent to sentry, and an error hash is returned with a meaningful value, and the parser returns successfully. this is similar to the previous test, but relies on us to determine when there is an error and actually returns an error message (I don't know why we didn't do that before)
2. raise an exception from the HTTP request, and test our handling. we'll need to understand what exceptions we expect to be raised, which involves looking at documentation for Net::HTTP that we're using to make the request. I know `Net::ReadTimeout` is an example, so we could force Net::HTTP to raise that exception and test our handling of it (at the least, exit gracefully)

I think there are examples in the code base of both approaches, though may be in non-parser classes. happy to talk more about them if needed!

Reply:
I went through this one again and I understood it better. When we get the twitter_metadata it looks  for the `twitter:creator` and tries to construct the author_url, if there isn't a twitter:creator it shouldn't error.

So I changed it to this
```
test "shouldn't error when cannot get twitter author url" do
    
  Parser::PageItem.stubs(:twitter_author_url).returns(nil)
    
  data = Parser::PageItem.new('https://random-page.com/page-item').parse_data(empty_doc)

  assert data['error'].nil?
end
```